agent/internal/client/base_client.go

```diff
@@ -372,109 +372,57 @@ func (c *baseCommunicator) GetLoggerProducer(ctx context.Context, tsk *task.Task
 			Task:   []LogOpts{{Sender: model.EvergreenLogSender}},
 		}
 	}
-	underlying := []send.Sender{}
 
-	exec, senders, err := c.makeSender(ctx, tsk, config.Agent, config, taskoutput.TaskLogTypeAgent)
+	exec, err := c.makeSender(ctx, tsk, config, taskoutput.TaskLogTypeAgent)
 	if err != nil {
 		return nil, errors.Wrap(err, "making agent logger")
 	}
-	underlying = append(underlying, senders...)
-	task, senders, err := c.makeSender(ctx, tsk, config.Task, config, taskoutput.TaskLogTypeTask)
+	task, err := c.makeSender(ctx, tsk, config, taskoutput.TaskLogTypeTask)
 	if err != nil {
 		return nil, errors.Wrap(err, "making task logger")
 	}
-	underlying = append(underlying, senders...)
-	system, senders, err := c.makeSender(ctx, tsk, config.System, config, taskoutput.TaskLogTypeSystem)
+	system, err := c.makeSender(ctx, tsk, config, taskoutput.TaskLogTypeSystem)
 	if err != nil {
 		return nil, errors.Wrap(err, "making system logger")
 	}
-	underlying = append(underlying, senders...)
 
 	return &logHarness{
-		execution:                 logging.MakeGrip(exec),
-		task:                      logging.MakeGrip(task),
-		system:                    logging.MakeGrip(system),
-		underlyingBufferedSenders: underlying,
+		execution: logging.MakeGrip(exec),
+		task:      logging.MakeGrip(task),
+		system:    logging.MakeGrip(system),
 	}, nil
 }
 
-func (c *baseCommunicator) makeSender(ctx context.Context, tsk *task.Task, opts []LogOpts, config *LoggerConfig, logType taskoutput.TaskLogType) (send.Sender, []send.Sender, error) {
+func (c *baseCommunicator) makeSender(ctx context.Context, tsk *task.Task, config *LoggerConfig, logType taskoutput.TaskLogType) (send.Sender, error) {
 	levelInfo := send.LevelInfo{Default: level.Info, Threshold: level.Debug}
 	var senders []send.Sender
 	if config.SendToGlobalSender {
 		senders = append(senders, grip.GetSender())
 	}
-	underlyingBufferedSenders := []send.Sender{}
```

Comment:
should we remove [this part](https://github.com/evergreen-ci/evergreen/blob/main/agent/internal/client/logger_producer.go#L69) as well?

Reply:
good catch!